pyiceberg/io/pyarrow.py

```diff
@@ -350,7 +351,7 @@ def parse_location(location: str) -> Tuple[str, str, str]:
             return uri.scheme, uri.netloc, f"{uri.netloc}{uri.path}"
 
     def _initialize_fs(self, scheme: str, netloc: Optional[str] = None) -> FileSystem:
-        if scheme in {"s3", "s3a", "s3n"}:
+        if scheme in {"s3", "s3a", "s3n", "oss", "r2"}:
```

Comment:
what is `oss`? I've never heard of it. And does S3FileSystem support both oss and r2?

Reply:
`oss` is protocol for Alibaba Cloud Object Storage Service, and it's [compatible with S3 API](https://www.alibabacloud.com/help/en/oss/developer-reference/use-amazon-s3-sdks-to-access-oss) as long as the URL is in virtual address style. 
![Screenshot 2024-12-03 093309](https://github.com/user-attachments/assets/941e3415-22f0-4c18-b66d-e6f806d8839f)
From my quick test it looks like the class does support OSS, but not sure yet about R2. Should I remove `r2` for now?

Reply:
Plugging out R2. [It uses account ID](https://developers.cloudflare.com/r2/api/s3/tokens/) to create the URL, which is quite different from how S3 natively set the virtual hosted style.

Reply:
thanks for testing it out! do you mind editing the PR description and mention this PR only supports oss